src/app/admin/admin-campaign-item/admin-campaign-item.component.ts

```diff
@@ -0,0 +1,42 @@
+import {ChangeDetectionStrategy, Component, Input, OnInit} from '@angular/core'
+import {CampaignService, CampaignWithInfo} from '../../shared/services/blockchain/campaign.service'
+import {AssetWithInfo} from '../../shared/services/blockchain/asset.service'
+import {FtAssetWithInfo} from '../../shared/services/blockchain/ft-asset.service'
+
+@Component({
+  selector: 'app-admin-campaign-item',
+  templateUrl: './admin-campaign-item.component.html',
+  styleUrls: ['./admin-campaign-item.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class AdminCampaignItemComponent implements OnInit {
+  @Input() asset!: AssetWithInfo | FtAssetWithInfo
+  @Input() campaign!: CampaignWithInfo
+  @Input() type!: 'list-item' | 'detail'
+
+  campaignData!: CampaignData
+
+  constructor(private campaignService: CampaignService,) {
+  }
+
+  ngOnInit(): void {
+    const stats = this.campaignService.stats(this.campaign)
+    const location = window.location
+    const issuerPath = location.pathname.split('/admin')[0]
+    const campaignUrl = `${location.protocol}//${location.host}${issuerPath}/offers/${this.campaign.ansName}`
```

Comment:
1. I think that `location.origin` combines both protocol and host.
2. you can just use `location.origin + issuerPathPipe.transform('/offers/${this.campaign.ansName}')` to calculate a campaign URL.
3. you can use location as a DI service (`private location: Location`)

Reply:
All good except for point 3: there is no provider for `Location` type.

Reply:
> All good except for point 3: there is no provider for `Location` type.

You can use window as DI and then call `window.location`. Does it work?